packages/cli/src/project.ts

```diff
@@ -0,0 +1,797 @@
+      }
+    }
+    return sourceInfos
+  }
+
+  private static async loadSourceFiles(projectRootDir: string, contractsRootDir: string): Promise<SourceInfo[]> {
+    const importsCache: string[] = []
+    const sourceInfos: SourceInfo[] = []
+    const loadDir = async function (dirPath: string): Promise<void> {
+      const dirents = await fsPromises.readdir(dirPath, { withFileTypes: true })
+      for (const dirent of dirents) {
+        if (dirent.isFile()) {
+          const sourcePath = path.join(dirPath, dirent.name)
+          const infos = await Project.loadSourceFile(projectRootDir, contractsRootDir, sourcePath, importsCache, false)
+          sourceInfos.push(...infos)
+        } else {
+          const newPath = path.join(dirPath, dirent.name)
+          await loadDir(newPath)
+        }
+      }
+    }
+    await loadDir(contractsRootDir)
+    const contractAndScriptSize = sourceInfos.filter(
+      (f) => f.type === SourceKind.Contract || f.type === SourceKind.Script
+    ).length
+    if (sourceInfos.length === 0 || contractAndScriptSize === 0) {
+      throw new Error('Project have no source files')
+    }
+    return sourceInfos.sort((a, b) => a.type - b.type)
+  }
+
+  static readonly DEFAULT_CONTRACTS_DIR = 'contracts'
+  static readonly DEFAULT_ARTIFACTS_DIR = 'artifacts'
+
+  static async build(
+    compilerOptionsPartial: Partial<CompilerOptions> = {},
+    projectRootDir = '.',
+    contractsRootDir = Project.DEFAULT_CONTRACTS_DIR,
+    artifactsRootDir = Project.DEFAULT_ARTIFACTS_DIR,
+    defaultFullNodeVersion: string | undefined = undefined,
+    skipSaveArtifacts = false
+  ): Promise<void> {
+    const provider = web3.getCurrentNodeProvider()
+    const fullNodeVersion = defaultFullNodeVersion ?? (await provider.infos.getInfosVersion()).version
+    const sourceFiles = await Project.loadSourceFiles(projectRootDir, contractsRootDir)
+    const { errorOnWarnings, ...nodeCompilerOptions } = { ...DEFAULT_COMPILER_OPTIONS, ...compilerOptionsPartial }
+    const projectArtifact = await ProjectArtifact.from(projectRootDir)
+    const changedSources = projectArtifact?.getChangedSources(sourceFiles) ?? sourceFiles
+    if (
+      projectArtifact === undefined ||
+      projectArtifact.needToReCompile(nodeCompilerOptions, fullNodeVersion) ||
+      changedSources.length > 0
+    ) {
+      if (fs.existsSync(artifactsRootDir)) {
+        removeOldArtifacts(artifactsRootDir, sourceFiles)
+      }
+      console.log(`Compiling contracts in folder "${contractsRootDir}"`)
+      Project.currentProject = await Project.compile(
+        fullNodeVersion,
+        provider,
+        sourceFiles,
+        projectRootDir,
+        contractsRootDir,
+        artifactsRootDir,
+        errorOnWarnings,
+        nodeCompilerOptions,
+        changedSources,
+        skipSaveArtifacts
+      )
+    }
+    // we need to reload those contracts that did not regenerate bytecode
+    Project.currentProject = await Project.loadArtifacts(
```

Comment:
Do we still need `Project.currentProject`?

Reply:
No, we don't need it anymore, removed